modules/dop/endpoints/project.go

```diff
@@ -681,3 +681,158 @@ func (e *Endpoints) ParseProjectTemplate(ctx context.Context, r *http.Request, v
+	if exportReq.OrgID == 0 || exportReq.ProjectID == 0 {
+		return apierrors.ErrExportProjectPackage.InvalidParameter(fmt.Errorf("orgID and projectID can't be empty")).ToResp(), nil
+	}
+	artifacts := []apistructs.Artifact{}
+	if err := json.NewDecoder(r.Body).Decode(&artifacts); err != nil {
+		return apierrors.ErrExportProjectPackage.InvalidParameter(fmt.Errorf("invalid artifacts: %v", err)).ToResp(), nil
+	}
+	if len(artifacts) == 0 {
+		return apierrors.ErrExportProjectPackage.InvalidParameter(fmt.Errorf("artifacts can't be empty")).ToResp(), nil
+	}
+	exportReq.Artifacts = artifacts
+
+	if !identityInfo.IsInternalClient() {
+		access, err := e.bdl.CheckPermission(&apistructs.PermissionCheckRequest{
+			UserID:   identityInfo.UserID,
+			Scope:    apistructs.OrgScope,
+			ScopeID:  uint64(exportReq.OrgID),
+			Resource: apistructs.ProjectPackageResource,
+			Action:   apistructs.GetAction,
+		})
+		if err != nil {
+			return apierrors.ErrExportProjectPackage.InternalError(err).ToResp(), nil
+		}
+		if !access.Access {
+			return apierrors.ErrExportProjectPackage.AccessDenied().ToResp(), nil
+		}
+	}
+	org, err := e.bdl.GetOrg(orgID)
+	if err != nil {
+		return apierrors.ErrExportProjectPackage.InternalError(err).ToResp(), nil
+	}
+	exportReq.OrgName = org.Name
+	project, err := e.bdl.GetProject(exportReq.ProjectID)
+	if err != nil {
+		return apierrors.ErrExportProjectPackage.InternalError(err).ToResp(), nil
+	}
+	if project.OrgID != uint64(exportReq.OrgID) {
+		return apierrors.ErrImportProjectPackage.InvalidParameter("projectID").ToResp(), nil
+	}
+	exportReq.ProjectName = project.Name
+	exportReq.ProjectDisplayName = project.DisplayName
+
+	fileID, err := e.project.ExportPackage(exportReq)
+	if err != nil {
+		return apierrors.ErrExportProjectPackage.InternalError(err).ToResp(), nil
+	}
+
+	ok, _, err := e.testcase.GetFirstFileReady(apistructs.FileProjectPackageExport)
+	if err != nil {
+		return errorresp.ErrResp(err)
+	}
+	if ok {
+		e.ExportChannel <- fileID
+	}
+
+	return httpserver.OkResp(fileID)
+}
+
+func (e *Endpoints) ImportProjectPackage(ctx context.Context, r *http.Request, vars map[string]string) (httpserver.Responser, error) {
+	identityInfo, err := user.GetIdentityInfo(r)
+	if err != nil {
+		return apierrors.ErrImportProjectPackage.NotLogin().ToResp(), nil
+	}
+	var req apistructs.ImportProjectPackageRequest
+	projectID, err := e.getProjectID(vars)
+	if err != nil {
+		return apierrors.ErrImportProjectPackage.InvalidParameter("projectID").ToResp(), nil
+	}
+	req.ProjectID = projectID
+	orgID, err := e.getOrgID(vars)
+	if err != nil {
+		return apierrors.ErrImportProjectPackage.InvalidParameter("orgID").ToResp(), nil
+	}
+	req.OrgID = orgID
+	req.IdentityInfo = identityInfo
+	if req.ProjectID == 0 {
+		return apierrors.ErrImportProjectPackage.InvalidParameter("projectID").ToResp(), nil
+	}
+	if req.OrgID == 0 {
+		return apierrors.ErrImportProjectPackage.InvalidParameter("orgID").ToResp(), nil
+	}
```

Comment:
You can add a func to get orgID / projectID / IdentityInfo.
Duplicate code in many places